value_test.go

```diff
@@ -935,3 +935,51 @@ func BenchmarkReadWrite(b *testing.B) {
 		}
 	}
 }
+
+// Regression test for https://github.com/dgraph-io/badger/issues/817
+func TestValueLogTruncate(t *testing.T) {
+	dir, err := ioutil.TempDir(".", "badger-test")
+	y.Check(err)
+	defer os.RemoveAll(dir)
+
+	opts := DefaultOptions
+	opts.Dir = dir
+	opts.ValueDir = dir
+	opts.Truncate = true
+
+	db, err := Open(opts)
+	require.NoError(t, err)
+	// Insert 1 entry so that we have valid data in first vlog file
+	require.NoError(t, db.Update(func(txn *Txn) error {
+		require.NoError(t, txn.Set([]byte("foo"), nil))
```

Comment:
you can simply return `txn.Set([]byte("foo", nil)`

Reply:
Fixed